features/step_definitions/barclamp_rabbitmq/rabbitmq_steps.rb

```diff
@@ -0,0 +1,74 @@
+Given(/^the barclamp proposal is using rabbitmq as messaging backend$/) do
+    json_response = JSON.parse(admin_node.exec!("crowbar rabbitmq show default").output)
+    expect(json_response["attributes"]["rabbitmq"]).not_to be_empty
+end
+
+When(/^the node with rabbitmq-server role has been detected successfully$/) do
+    @node = nodes.find(barclamp: "rabbitmq", element: "rabbitmq-server").first
```

Comment:
Yes. I dont know that for sure. So just added a netstat check to ensure the port 5672 is listening actively on the node before fixing the node to test on. I verified that the command shows success=false for passive node and true for active node. Hope this would work for most cases.